### PR TITLE
docs(changelog): add 2.2.0 changes

### PR DESCRIPTION
**Pull request by mr-cal**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

Lots of features in 2.2.0!

### CODE REVIEW

**Review by mattculler — APPROVED** [review]

**Review by bepri — APPROVED** [review]

**Comment by lengau** [comment]
Any chance we can get https://github.com/canonical/craft-parts/pull/940/files into 2.2.0?

**Comment by mr-cal** [comment]
> Any chance we can get https://github.com/canonical/craft-parts/pull/940/files into 2.2.0?

Yes, I can wait for that.

**Comment by lengau** [comment]
> Yes, I can wait for that.

Thanks! I've added the changelog entry for that there.

File: docs/changelog.rst [hunk 1]
```diff
@@ -2,21 +2,45 @@
...
 ------------------
 
 New features:
 
-- Add a :doc:`uv plugin</common/craft-parts/reference/plugins/uv_plugin>` for
-  projects that use the uv build system.
+- Add a :ref:`uv plugin<craft_parts_uv_plugin>` for projects that use the `uv
+  <https://docs.astral.sh/uv/>`_ build system.
+- Add a :ref:`Go Use plugin<craft_parts_go_use_plugin>` for setting up a `go
+  workspace` for Go modules.
```
Comment (medubelko) [inline docs/changelog.rst:13]:
```suggestion
- Add a :ref:`Go Use plugin<craft_parts_go_use_plugin>` for setting up a `
  `workspace <https://go.dev/ref/mod#workspaces>`_ for Go modules.
```

File: docs/changelog.rst [hunk 2]
```diff
@@ -2,21 +2,45 @@
...
+- Add an API for :ref:`registering custom source types
+  <how_to_add_a_source_handler>`.
+- Prefer ``craft.git`` as the binary to handle git sources, in environments
+  where it is available.
```
Comment (medubelko) [inline docs/changelog.rst:17]:
```suggestion
- Prefer ``craft.git`` as the binary to handle Git sources, in environments
  where it's available.
```

File: docs/changelog.rst [hunk 3]
```diff
@@ -2,21 +2,45 @@
...
+- Set ``JAVA_HOME`` environment variable in Java-based plugins. The plugin will
+  try to detect the latest available JDK.
+- Add ``part_has_slices`` function to determine if a part has slices in its
+  ``stage-packages`` key.
+- Add ``part_has_chisel_as_build_snap`` function to determine if a part lists
+  ``chisel`` as a ``build-snap``.
```
Comment (medubelko) [inline docs/changelog.rst:23]:
```suggestion
- Add a ``part_has_slices`` function to determine if a part has slices in its
  ``stage-packages`` key.
- Add a ``part_has_chisel_as_build_snap`` function to determine if a part
  lists ``chisel`` as a ``build-snap``.
```

File: docs/changelog.rst [hunk 4]
```diff
@@ -2,21 +2,45 @@
...
+- Add ``chisel`` as a ``build-snap`` if any part has slices and ``chisel`` is
+  not already listed as a ``build-snap``.
```
Comment (medubelko) [inline docs/changelog.rst:25]:
```suggestion
- Add ``chisel`` as a ``build-snap`` if any part has slices and ``chisel``
  isn't already listed as a ``build-snap``.
```

File: docs/changelog.rst [hunk 5]
```diff
@@ -2,21 +2,45 @@
...
+- Fix dependency validation for the ``rust`` plugin when a ``rust-deps`` part
+  exists.
+
+Documentation:
+
+- Add labels to the :ref:`ant plugin<craft_parts_ant_plugin>` and
+  :ref:`maven plugin<craft_parts_maven_plugin>` reference pages.
+- Add a link to common part properties from the :ref:`npm
+  plugin<craft_parts_npm_plugin>` reference page.
+- Add links to commit history for ``2.x`` releases in the changelog.
```
Comment (medubelko) [inline docs/changelog.rst:41]: Personally, I find this sounds like we're praising a house for not falling down.

**Review by medubelko — CHANGES_REQUESTED** [review]
All nitpicks except for one missing link.

File: docs/changelog.rst (hunk 2 above)
Comment (mr-cal) [reply docs/changelog.rst:17]: I don't know why I keep avoiding contractions in changelogs.

File: docs/changelog.rst (hunk 5 above)
Comment (mr-cal) [reply docs/changelog.rst]: Heh, I'm just celebrating the wins.  (I removed it)

**Review by medubelko — APPROVED** [review]

**Pull request merged**